src/target/cortexa.c

```diff
@@ -546,10 +570,10 @@ bool cortexa_attach(target_s *target)
 	target_check_error(target);
 
 	/* Enable halting debug mode */
-	uint32_t dbgdscr = apb_read(target, DBGDSCR);
-	dbgdscr |= DBGDSCR_HDBGEN | DBGDSCR_ITREN;
-	dbgdscr = (dbgdscr & ~DBGDSCR_EXTDCCMODE_MASK) | DBGDSCR_EXTDCCMODE_STALL;
-	apb_write(target, DBGDSCR, dbgdscr);
+	uint32_t dbgdscr = cortex_dbg_read32(target, CORTEXAR_DBG_DSCR);
+	dbgdscr |= CORTEXAR_DBG_DSCR_ITR_ENABLE | CORTEXAR_DBG_DSCR_ITR_ENABLE;
```

Comment:
to be fixed in PR1628, should be `CORTEXAR_DBG_DSCR_HALT_DBG_ENABLE | CORTEXAR_DBG_DSCR_ITR_ENABLE`